src/openzaak/components/zaken/api/validators.py

```diff
@@ -525,3 +527,71 @@ def __call__(self, attrs, serializer):
 
         if zaak.archiefstatus != Archiefstatus.nog_te_archiveren:
             raise serializers.ValidationError(self.message, code=self.code)
+
+
+def match_eigenschap_specificatie(spec, value: str) -> bool:
+    """
+    validate value against eigenschap.specificatie
+    moved to the separate function to reuse for admin validation
+    """
+    # enum
+    if spec.waardenverzameling:
+        return value in spec.waardenverzameling
+
+    if spec.formaat == FormaatChoices.tekst:
+        return len(value) <= int(spec.lengte)
+
+    if spec.formaat == FormaatChoices.getal:
+        whole_length = spec.lengte.split(",")[0]
+        regex = rf"\d{{0,{whole_length}}}"
+        if "," in spec.lengte:
+            fractional_length = spec.lengte.split(",")[-1]
+            regex += rf",?\d{{0,{fractional_length}}}"
+
+        match = re.fullmatch(regex, value)
+        return bool(match)
+
+    if spec.formaat == FormaatChoices.datum:
+        # according ZGW standard datum should be in 'jjjjmmdd' format
+        try:
+            datetime.strptime(value, "%Y%m%d")
+        except ValueError:
+            return False
+        else:
+            return True
+
+    if spec.formaat == FormaatChoices.datum_tijd:
+        # according ZGW standard datum/tijd should be in 'jjjjmmdduummss' format
+        try:
+            datetime.strptime(value, "%Y%m%d%H%M%S")
+        except ValueError:
+            return False
+        else:
+            return True
+
+    return True
+
+
+class ZaakEigenschapValueValidator:
+    code = "waarde-incorrect-format"
+    message = _("The 'waarde' value doesn't match the related eigenschap.specificatie")
+    requires_context = True
+
+    def __call__(self, attrs, serializer):
+        if not settings.ZAAK_EIGENSCHAP_WAARDE_VALIDATION:
```

Comment:
Not really a suggestion, but I wonder if you can conditionally add it to the serializer validators so it does not check this every time.

I am not sure if this is considered DRF bad practice.

Reply:
I haven't seen such thing in the docs. I think it's expected that there is a list of validators for each field and the conditions are put inside the validator. 
In this particular case I don't think that it will win a lot of performance, since we don't do any extra DB or HTTP requests before the condition. 

And even if it happens the way to go is to cache something on the serializer level and then use context to pass it to the validator